src/emqx_mqueue.erl

```diff
@@ -121,14 +121,15 @@ stats(#mqueue{max_len = MaxLen, dropped = Dropped} = MQ) ->
 -spec(in(message(), mqueue()) -> {maybe(message()), mqueue()}).
 in(#message{qos = ?QOS_0}, MQ = #mqueue{store_qos0 = false}) ->
     {_Dropped = undefined, MQ};
-in(Msg = #message{topic = Topic}, MQ = #mqueue{default_p = Dp,
+in(Msg = #message{topic = Topic, qos = Qos}, MQ = #mqueue{default_p = Dp,
                                                p_table = PTab,
                                                q = Q,
                                                len = Len,
                                                max_len = MaxLen,
                                                dropped = Dropped
                                               } = MQ) ->
-    Priority = get_priority(Topic, PTab, Dp),
+    Qos_ = integer_to_binary(Qos),
+    Priority = get_priority(<<Topic/binary, $/, Qos_/binary>>, PTab, Dp),
```

Comment:
Why are you adding `<<$/, Qos_/binary>>` here? The docs in schema do not talk about the QoS. Also, using the separator `/` wouldn't be a good idea, the config gets difficult to understand that way.

What is the goal of separating the queues by QoS? How will this be exposed in the config nicely?

Reply:
Hi,@phanimahesh

emqx.conf 
...
## Topic priorities.
 ## 'none' to indicate no priority table (by default), hence all messages
 ## are treated equal
 ##
 ## Priority number [1-255]
 ## Example: topic/1=10,topic/2=8
 ## NOTE: comma and equal signs are not allowed for priority topic names
 ## NOTE: messages for topics not in the priority table are treated as
 ##       either highest or lowest priority depending on the configured
 ##       value for mqueue_default_priority
 ##
 zone.external.mqueue_priorities = topic/1=10,topic/2=8

Reply:
It is not clear that those are the QoS. It looks like the example is talking about two topics, `topic/1` and `topic/2`.

Reply:
@phanimahesh , It looks like I got it wrong, thanks!

Reply:
Maybe we should ask @emqplus for his opinion here?

Reply:
I think you are right, using the separator / to separate qos wouldn't be a good idea, this confuses the topic and qos.